python/langsmith/client.py

```diff
@@ -332,93 +347,47 @@ def request_with_retries(
         request_method: str,
         url: str,
         request_kwargs: Mapping,
-    ) -> requests.Response:
-        """Send a request with retries.
-
-        Parameters
-        ----------
-        request_method : str
-            The HTTP request method.
-        url : str
-            The URL to send the request to.
-        request_kwargs : Mapping
-            Additional request parameters.
-
-        Returns
-        -------
-        Response
-            The response object.
+        filter_null_params: bool = True,
+    ) -> httpx.Response:
+        """Send a request with retries."""
+        with _request_error_handler(request_method, url) as handler_state:
```

Comment:
Thoughts on this error handling context manager to reduce duplicate handling code?